src/mono/mono/metadata/appdomain.c

```diff
@@ -1054,7 +1054,7 @@ mono_domain_owns_vtable_slot (MonoDomain *domain, gpointer vtable_slot)
 	gboolean res;
 
 	mono_domain_lock (domain);
-	res = mono_mempool_contains_addr (domain->mp, vtable_slot);
```

Comment:
Does't this need to lock on the memory manager ?

Reply:
Yes, oops.